addons/sale/models/sale.py

```diff
@@ -612,6 +612,19 @@ def _compute_qty_delivered_updateable(self):
         for line in self:
             line.qty_delivered_updateable = (line.order_id.state == 'sale') and (line.product_id.track_service == 'manual') and (line.product_id.expense_policy == 'no')
 
+    @api.depends('invoice_lines', 'invoice_lines.invoice_id', 'invoice_lines.invoice_id.state', 
+                 'invoice_lines.invoice_id.refund_invoice_ids.invoice_line_ids', 'invoice_lines.invoice_id.refund_invoice_ids.invoice_line_ids.invoice_id', 'invoice_lines.invoice_id.refund_invoice_ids.invoice_line_ids.invoice_id.state')
+    def _compute_invoice_amount(self):
+        for line in self:
+            amt_invoiced = 0.0
+            refund_amount_total = 0.0
+            for invoice_line in line.invoice_lines:
+                if invoice_line.invoice_id.state in ('open', 'paid'):
+                    amt_invoiced += invoice_line.price_total
+                refund_amount_total += sum(invoice_line.invoice_id.refund_invoice_ids.filtered(lambda x: x.state in ['open', 'paid']).mapped('amount_total'))
+            line.amt_invoiced = amt_invoiced - refund_amount_total
+            line.amt_to_invoice = line.price_total - amt_invoiced
+
```

Comment:
It look strange to mix Python Built-in functions with a for loop for this kind of computation.
I think it's more clear something like:

```
invoices = line.invoice_lines.mapped('invoice_id').filtered(lambda inv: inv.state in ('open', 'paid'))
refund_invoices = invoices.mapped('refund_invoice_ids').filtered(lambda inv: inv.state in ('open', 'paid'))
invoiced_amount_total += sum(invoices.mapped('price_total'))
refund_amount_total = sum(refund_invoices.mapped('amount_total'))
```